superset/utils/cache_manager.py

```diff
@@ -15,15 +15,40 @@
 # specific language governing permissions and limitations
 # under the License.
 import logging
+from typing import Any, Optional, Union
 
 from flask import Flask
 from flask_caching import Cache
+from markupsafe import Markup
+
+from superset.utils.core import DatasourceType
 
 logger = logging.getLogger(__name__)
 
 CACHE_IMPORT_PATH = "superset.extensions.metastore_cache.SupersetMetastoreCache"
 
 
+class ExploreFormDataCache(Cache):
+    def get(self, *args: Any, **kwargs: Any) -> Optional[Union[str, Markup]]:
+        cache = self.cache.get(*args, **kwargs)
+
+        if not cache:
+            return None
+
+        # rename data keys for existing cache based on new TemporaryExploreState model
+        if isinstance(cache, dict):
```

Comment:
Why do we need to check if the cache is a dict? Won't it always be a dict?

Reply:
actually, some times it returns a string.

Reply:
superset/explore/form_data/commands/create.py line 52 for example, it returns the key, not the data. I'm honestly not sure why.